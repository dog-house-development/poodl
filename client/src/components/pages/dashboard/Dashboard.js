import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { logoutUser } from "../../../actions/authActions";
import Button from "../../inputs/Button";

const propTypes = {
  logoutUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired
};

export class Dashboard extends Component {
  onLogoutClick = e => {
    e.preventDefault();
    this.props.logoutUser();
  };

  render() {
    const { user } = this.props.auth;

    return (
      <div>
        <h4>
          <p>Hey there, {user.name.split(" ")[0]}. You are now logged in</p>
        </h4>
        <Button content="log out" onClick={this.onLogoutClick} />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.auth
});

Dashboard.propTypes = propTypes;
export default connect(
  mapStateToProps,
  { logoutUser }
)(Dashboard);
