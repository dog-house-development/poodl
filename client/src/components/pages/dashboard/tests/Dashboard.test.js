import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';

import { Dashboard, mapStateToProps, mapDispatchToProps } from '../Dashboard';

configure({ adapter: new Adapter() });

describe('Dashbaord tests', () => {
    let state, props, wrapper, instance;
    const setInstanceAndWrapper = (_props = {}, _state = {}) => {
        state = _.assign(
            {},
            {
                auth: {
                    isAuthenticated: true,
                    loading: false,
                    admin: {
                        id: '5c52379d9be6fc0017afd46e',
                        name: 'Sandwich Man'
                    }
                },
                errors: {}
            },
            _state
        );
        props = _.assign({}, _props);
        wrapper = shallow(
            <Dashboard
                {..._.assign(
                    {},
                    props,
                    mapStateToProps(state, props),
                    mapDispatchToProps(jasmine.createSpy('dispatch'))
                )}
            />
        );
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('mapStateToProps', () => {
        it('should map state to props', () => {
            expect(mapStateToProps(state, props)).toEqual({
                auth: {
                    isAuthenticated: true,
                    loading: false,
                    admin: {
                        id: '5c52379d9be6fc0017afd46e',
                        name: 'Sandwich Man'
                    }
                }
            });
        });
    });

    describe('mapDispatchToProps', () => {
        it('should map dispatch to props', () => {
            const dispatch = jest.fn();
            expect(JSON.stringify(mapDispatchToProps(dispatch))).toEqual(JSON.stringify({ logoutAdmin: () => {} }));
        });
    });

    describe('onLogoutClick', () => {
        it('should run without errors', () => {
            spyOn(instance, 'onLogoutClick');
            const e = {
                target: { name: 'email', value: 'Moo' },
                preventDefault: () => {}
            };
            wrapper.find('Button').simulate('click', e);
        });
    });

    describe('render', () => {
        it('should render correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
