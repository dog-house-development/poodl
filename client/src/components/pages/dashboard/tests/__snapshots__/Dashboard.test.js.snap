// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashbaord tests render should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    auth={
      Object {
        "admin": Object {
          "id": "5c52379d9be6fc0017afd46e",
          "name": "Sandwich Man",
        },
        "isAuthenticated": true,
        "loading": false,
      }
    }
    logoutAdmin={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Hey there, 
        </h2>,
        <h1>
          Sandwich
          .
        </h1>,
        <p>
          You are now logged in
        </p>,
        <Button
          content="Log out"
          kind="primary"
          onClick={[Function]}
          size="medium"
        />,
      ],
      "className": "dashboard-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hey there, ",
        },
        "ref": null,
        "rendered": "Hey there, ",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Sandwich",
            ".",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Sandwich",
          ".",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You are now logged in",
        },
        "ref": null,
        "rendered": "You are now logged in",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "content": "Log out",
          "kind": "primary",
          "onClick": [Function],
          "size": "medium",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Hey there, 
          </h2>,
          <h1>
            Sandwich
            .
          </h1>,
          <p>
            You are now logged in
          </p>,
          <Button
            content="Log out"
            kind="primary"
            onClick={[Function]}
            size="medium"
          />,
        ],
        "className": "dashboard-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hey there, ",
          },
          "ref": null,
          "rendered": "Hey there, ",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Sandwich",
              ".",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Sandwich",
            ".",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You are now logged in",
          },
          "ref": null,
          "rendered": "You are now logged in",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "content": "Log out",
            "kind": "primary",
            "onClick": [Function],
            "size": "medium",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
